src/java.desktop/share/classes/javax/swing/JComponent.java

```diff
@@ -4191,25 +4191,25 @@ void clientPropertyChanged(Object key, Object oldValue,
      * @param value Object containing the property value
      */
     void setUIProperty(String propertyName, Object value) {
-        if (propertyName == "opaque") {
+        if (propertyName.equals("opaque")) {
```

Comment:
This will throw an NPE if `propertyName` is `null`, which didn't happened before the fix.

> this method could be rewritten to use strings in switch. 

Probably we might still want to use switch, it will handle the NPE case as before the fix.

Did you run CI tests after the fix?

Reply:
I have taken care of NPE..I am not sure of switch which probably might be a hindrance in backporting if needed to earlier release trains.
CI is also run..link in JBS.

Reply:
> I am not sure of switch which probably might be a hindrance in backporting if needed to earlier release trains.

You can still use the old switch statement with breaks.

Reply:
It will not be consistent if we use switch only in this place because in most places, we either use the way we have before fix or or what I have done in version 1 or what I have done now. Probably, we need a PR to change other places where we have use pre-fix ways.